### ISSUE TITLE
Story: Add External Link Icon to NCIDS Collections

### ISSUE THREAD

**Issue by andyvanavery31** [NCIOCPL/ncids#1900]
### Specifications

**Visual**

- External Link icon is center aligned vertically on the title text
- Follows usa-link styling
- If the title would wrap, the last word of the title will wrap with the icon so the icon does not sit alone

**Functional**
- If the link text does not have `.gov` it will get an external link icon
- A Collection Item can be overridden to have an external link icon
- A Collection Item with an external link icon can be overridden to remove it

**Impacted Components**

- Collection
  - [Doc Site](https://designsystem-dev.cancer.gov/develop/components/collection)
  - [Storybook Collection Default](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-collection-default--default)
  - [Storybook Collection Media](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-collection-default--media)
  - [Storybook Collection Calendar](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-collection-default--calendar)
  - [Storybook Collection Condensed](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-collection-variants--condensed)

### DoR Tasks

- [x] Comps Created
- [x] Comps Reviewed using [Checklist](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Product-Review-of-Design-Solutions)
  - [x] Developer Review
  - [x] IA Review
  - [x] Product Review
- [x] Comps Approved
  - [x] Developer Approval
  - [x] IA Approval
  - [x] Product Approval
- [x] Scenarios Created
  - [x] Picture for collection with media to be created and added to the ticket by designers
- [x] Scenarios Reviewed
  - [x] Designer Review
  - [x] Developer Review
  - [x] IA Review
- [x] Scenarios Approved
  - [x] Designer Approval
  - [x] Developer Approval
  - [x] IA Approval
- [x] Acceptance Criteria Drafted
- [x] Acceptance Criteria Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] Acceptance Criteria Approved
  - [x] Developer Approval
  - [x] IA Approval

### DoD Tasks

- [ ] Governance Drafted
- [ ] PR Review (see PR for review and comments)
- [ ] Client review with demo

### Technical Notes
**Scenarios/Comps**
- Use the comps here: [Default Collection](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=8690-6214&node-type=text&m=dev), [Collection with Media](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=8420-61624&node-type=instance&m=dev) for reference. Note: The Collections with Calendar and Condensed are not shown on comps, but the functionality is the same as the displayed versions.
- On the [Documentation Site Collection page](https://designsystem-dev.cancer.gov/develop/components/collection) add:
  - In the [Best Practices](https://designsystem-dev.cancer.gov/develop/components/collection#best-practices) section add a subsection which will be the last subsection with the heading, "External Links", and in the subsequent paragraph explain how and when to use and add the External Link Icon. In this section, explain how the external links are mostly "auto-magical", and provide an explanation of how to use the override.
  - A collection item to the bottom of the [Default collection code snippet](https://designsystem-dev.cancer.gov/develop/components/collection#default-collection) that is an external link. The title will be "External Link Example", date of, "November 15, 2024", description of "This collection item is an external link that goes to Google", with the link of www.google.com, and no tags.
  - A collection item to the bottom of the [Condensed collection code snippet](https://designsystem-dev.cancer.gov/develop/components/collection#condensed-collection) that is an external link. The title will be "External Link Example", with the displayed link of Google.com, and the link being www.google.com
  - A collection item to the bottom of the [Default with media code snippet](https://designsystem-dev.cancer.gov/develop/components/collection#default-with-media) that is an external link. The title will be "External Link Example", no author, date of November 15, 2024, description of "This collection item is an external link that goes to Google", with the link of www.google.com, and no tags. Add the image supplied by the designers added to this ticket.
  - A collection item to the bottom of the [Default with calendar code snippet](https://designsystem-dev.cancer.gov/develop/components/collection#default-with-calendar) that is an external link. The title will be "External Link Example", with the date of Nov 15, and description of "This collection item is an external link that goes to Google", with the link of www.google.com, and no tags
- On Storybook:
  - Within the Collections folder, add a folder called "Test Cases", and add four stories "Default with External Link", "Media with External Link", "Calendar with External Link", and "Condensed with External Link". For each of these stories, replicate the code snippet from the Doc Site as described above. In addition to replicating them, replace the third to last colleciton item with an external link, change the collection item title to "External Link Override", change the description of the collection item to "Example where the link is external, but the external link icon is overridden.", and use the override class to not display the external link icon. On the second to last colleciton item on each of these stories, change the collection item title to "Internal Link Override", change the description of the collection item to "Example where the link is internal, but it is overridden so the external link icon is displayed.", and use the override class to display the external link icon.

### Acceptance Criteria

```gherkin
Scenario: A collection with an external link is viewed on the NCIDS
  Given the external link icon has been added to the NCIDS Documentation Site and NCIDS Storybook for Collections
    And the collection item is linked to a url that does not contain `.gov`
  When the collection item has a linked url that does not contain `.gov`
  Then the collection item will appear with the external link icon
    And the icon on the collection item will match the comp (see links above)
    And the external link icon will remain center aligned vertically on the line of text it is on
  When the text of the collection item title wraps
    And the width of the collection item is only narrow enough where the external link icon should wrap alone
  Then the last word of the collection title will wrap to the next line with the external link icon
  When a user wants to override the behavior of a collection item with a link that does not contain `.gov` to not display the external link icon
  Then they will override the behavior with an override class
    And the external link icon will not be displayed
  When a user has a collection item with a link that contains `.gov`
    And they want to override the behavior to include the external link icon
  Then they will override the behavior with an override class
    And the external link icon will be displayed
```

**Comment by adrianacastaneda** [comment]
@andyvanavery31 Question on the following lines in the ACs. Let me know your thoughts! thanks!

"When a user wants to override the behavior of a **card** with a link that does not contain `.gov` to not display the external link icon. Then they will override the behavior with an override class. And the external link icon will not be displayed"
- this sounds right to me but why is it referred to as a card? The same applies for this line "When a user has a card with a link that contains `.gov`"

**Comment by blilianyu** [comment]
@andyvanavery31 
![media-collections-external-link](https://github.com/user-attachments/assets/52523584-fd14-466a-a123-52d1c5b53829)

**Comment by blilianyu** [comment]
Here are some images for testing.

![image-1](https://github.com/user-attachments/assets/65d5fdbb-124c-4b11-adba-b89d5451be2d)
![image-2](https://github.com/user-attachments/assets/83a39bee-8c11-476e-92c5-c2934616b02f)
![image-3](https://github.com/user-attachments/assets/f53e2b46-e9ac-407f-89e2-1abebd303cb7)
![image-4](https://github.com/user-attachments/assets/314e8c24-a829-43ef-a96a-888c652c1ede)

cc: @andyvanavery31
